PSSM_score_Peptide/main.cpp

```diff
@@ -162,6 +162,39 @@ void getFileNamesFromArgv(int argc, char *argv[], string & PSSM_FileName, string
 	cout<<endl;
 }
 
+void getFileNamesFromArgv(int argc, char *argv[], string & PSSM_FileName, string & CutofsPerPSSM_FileName, string & Seq_FASTA_FileName, string & Hits_Out_FileName, size_t & numberOfRandomPSSM, bool &use_factor) {
+	// parse ARGV arguments
+	size_t num_required_params = 6;
+	if (argc != (num_required_params * 2)+2) {// each with its flag and mode_flag, check the value of argc. If not enough parameters have been passed, inform user and exit.
+		cout << "Usage is -pssm <PSSMs_in_MAST_Format> -pssm_cutoffs <filename_for PSSM_cutoffs> -seq <input_seq_FASTA> -out <out>\n"; // Inform the user of how to use the program
+		exit(12);
+	}
+	cout << argv[0] <<" ";
+	for (int i = 1; i < argc; ++i) // iterate over flags
+	{
+		/* We will iterate over argv[] to get the parameters stored inside.
+		* Note that we're starting on 1 because we don't need to know the
+		* name of the program, which is stored in argv[0] */
+		if (string(argv[i]) == "-pssm") PSSM_FileName = string(argv[i + 1]);
+		else if (string(argv[i]) == "-pssm_cutoffs") CutofsPerPSSM_FileName = string(argv[i + 1]);
+		else if (string(argv[i]) == "-seq") Seq_FASTA_FileName = string(argv[i + 1]);
+		else if (string(argv[i]) == "-out") Hits_Out_FileName = string(argv[i + 1]);
+		else if (string(argv[i]) == "-NrandPSSM") numberOfRandomPSSM=size_t(atoi(argv[i + 1]));
+		else if (string (argv[i]) == "-useFactor"){
+			if (string(argv[i + 1]) == "True"){
```

Comment:
we are mixing python and C++ notations (True vs true).
Also it would be enough if flag exists, i.e. no value needed